### PR TITLE
backend/session/freebsd: Fix the way to get TTY path

### PR DESCRIPTION
**Pull request by lantw44**
Previously, the path of TTY is generated using snprintf with %d format.
It works with TTY 1 to 10, but fails with TTY with greater number
because the number used in the name is in base 32 instead of base 10.
Since there is no standard function to convert a number to a string with
a custom base, this commit adds a function to do it.

Fixes: https://github.com/swaywm/wlroots/issues/1854

### CODE REVIEW

File: backend/session/direct-freebsd.c [hunk 1]
```diff
@@ -139,6 +139,50 @@ static int vt_handler(int signo, void *data) {
...
+	int num = tty - 1;
+	if (num == 0) {
+		path[offset++] = '0';
+		path[offset++] = '\0';
+		return true;
+	}
+
+	const int base = 32;
+	size_t num_len = 0;
+	for (int remaning = num; remaning > 0; remaning /= base, num_len++);
```
Comment (emersion) [inline backend/session/direct-freebsd.c]: Instead we can just set this to a reasonable value.

File: backend/session/direct-freebsd.c [hunk 2]
```diff
@@ -139,6 +139,50 @@ static int vt_handler(int signo, void *data) {
...
+
+	// Return early if the buffer is too small.
+	if (prefix_len + num_len + 1 > len) {
+		return false;
+	}
+
+	char num_buf[num_len];
```
Comment (emersion) [inline backend/session/direct-freebsd.c]: No need for this intermediary buffer

File: backend/session/direct-freebsd.c (hunk 1 above)
Comment (lantw44) [reply backend/session/direct-freebsd.c]: Fixed by moving the length check into the loop.

File: backend/session/direct-freebsd.c (hunk 2 above)
Comment (lantw44) [reply backend/session/direct-freebsd.c]: Fixed by swapping bytes in-place.

**Pull request merged**